service/src/test/java/bio/terra/externalcreds/services/NihCredentialsSyncFunctionalTest.java

```diff
@@ -0,0 +1,78 @@
+package bio.terra.externalcreds.services;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+import bio.terra.externalcreds.BaseTest;
+import bio.terra.externalcreds.config.ExternalCredsConfig;
+import bio.terra.externalcreds.dataAccess.GoogleCloudStorageDAO;
+import bio.terra.externalcreds.util.EcmRestTemplate;
+import com.google.cloud.storage.BlobId;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+@Tag("functional")
+class NihCredentialsSyncFunctionalTest extends BaseTest {
+
+  @Autowired GoogleCloudStorageDAO googleCloudStorageDAO;
+  @Autowired ExternalCredsConfig externalCredsConfig;
+  @SpyBean NihCredentialsSyncService nihCredentialsSyncService;
+  @MockBean EcmRestTemplate restTemplate;
+
+  private final Set<String> allowlistNames =
+      Set.of(
+          "test-allowlist-1.txt",
+          "test-allowlist-2.txt",
+          "test-allowlist-3.txt",
+          "phs000004.c1.allowlist.txt");
+
+  @BeforeEach
+  public void setup() {
```

Comment:
Could we mock this out instead so that you don't have to actually read/write to google as part of the test?  You can still have a test somewhere that tests that your GoogleCloudStorageDAO can actually connect to GCP, but you don't need to test that functionality in these tests too.  

This sort of refactoring is exactly what we are going to start doing in Sam soon too, so it might be a good exercise to practice this here.